x11-libs/libva-intel-media-driver/libva-intel-media-driver-20.4.5.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+if [[ ${PV} == *9999 ]] ; then
+	: ${EGIT_REPO_URI:="https://github.com/intel/media-driver"}
+	if [[ ${PV%9999} != "" ]] ; then
+		: ${EGIT_BRANCH:="release/${PV%.9999}"}
+	fi
+	inherit git-r3
+else
+	SRC_URI="https://github.com/intel/media-driver/archive/intel-media-${PV}.tar.gz"
+	S="${WORKDIR}/media-driver-intel-media-${PV}"
+	KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="Intel Media Driver for VAAPI (iHD)"
+HOMEPAGE="https://github.com/intel/media-driver"
+
+LICENSE="MIT BSD"
+SLOT="0"
+IUSE="X set-as-default latest-cxx +user-cflags"
+
+PATCHES=( "${FILESDIR}"/${PN}-20.2.0_x11_optional.patch
+          "${FILESDIR}"/${PN}-20.4.5_user_cflags.patch
+        )
+
+DEPEND=">=media-libs/gmmlib-20.4.1
+	>=x11-libs/libva-2.10.0[X?]
+	>=x11-libs/libpciaccess-0.13.1-r1:=
+"
+RDEPEND="${DEPEND}"
+
+src_configure() {
+	local mycmakeargs=(
+		-DMEDIA_BUILD_FATAL_WARNINGS=OFF
+		-DMEDIA_RUN_TEST_SUITE=OFF
+		-DBUILD_TYPE=release
+		-DPLATFORM=linux
+		-DUSE_X11=$(usex X)
+		-DLATEST_CPP_NEEDED=$(usex latest-cxx)
+		-DOVERRIDE_COMPILER_FLAGS=$(usex !user-cflags)
```

Comment:
Normally we use _custom-cflags_ for this, but do we definitely need it? I'd prefer to just always use user CFLAGS but I guess this may affect support upstream.

Reply:
Thanks, I saw `custom-cflags` before, but I was unable to find it in Local USE flags. I missed the global flags. I will rename it.
Still want to make it optional (but enabled by default) as upstream didn't test such builds. I believe that we can have it enabled by default as I carefully selected which compiler flags to allow from user flags and which flags are allowed from package configuration. All warning and error flags are still defined by package (`-Wall -Wno-unused -Werror=address` etc.), as well as some code generation flags (`-fno-short-enums -Wa,--noexecstack -fno-strict-aliasing`), while optimisation flags, arch and debug flags are blocked from package configuration so user flags are used (`-O2 -g -march=corei7 -mtune=atom -m64` etc.) therefore it should be safe to use user-defined flags, but it case of any problems it worth to build the package with the upstream flags to check where problem is coming from.

Reply:
ack :)